src/nvim/ex_docmd.c

```diff
@@ -9411,3 +9411,40 @@ static void ex_terminal(exarg_T *eap)
     xfree(name);
   }
 }
+
+static void ex_plug(exarg_T *eap)
+{
+  char *path = NULL;
+  if (*eap->arg != NUL) {
+    if (os_isdir(eap->arg)) {  // full path: plug ~/.config/nvim/plugged/coolplugin
```

Comment:
90% that this is not going to work. `~/` will not be expanded, so `~/.config/nvim/plugged/coolplugin` will be recognized as a directory only if there is directory `./~/.config/nvim/plugged/coolplugin`.

100% that this is completely incorrect because `plug coolplugin` will use this branch should `coolplugin` directory be contained in the current directory, no matter the intent.